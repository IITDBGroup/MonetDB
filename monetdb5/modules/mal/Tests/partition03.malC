# function user.s6_1():void;                                                                                                         |
#     X_1:void := querylog.define("explain select i, count(*) from tmp2 group by i;":str, "minimal_pipe":str, 29:int);               |
#     X_4:int := sql.mvc();                                                                                                          |
#     C_5:bat[:oid] := sql.tid(X_4:int, "sys":str, "tmp2":str);                                                                      |
#     X_8:bat[:int] := sql.bind(X_4:int, "sys":str, "tmp2":str, "i":str, 0:int);                                                     |
#     X_11:bat[:int] := sql.emptybind(X_4:int, "sys":str, "tmp2":str, "i":str, 1:int);                                               |
#     (C_13:bat[:oid], X_14:bat[:int]) := sql.emptybind(X_4:int, "sys":str, "tmp2":str, "i":str, 2:int);                             |
#     X_16:bat[:int] := sql.delta(X_8:bat[:int], C_13:bat[:oid], X_14:bat[:int], X_11:bat[:int]);                                    |
#     X_17:bat[:int] := algebra.projection(C_5:bat[:oid], X_16:bat[:int]);                                                           |
#     (X_18:bat[:oid], C_19:bat[:oid], X_20:bat[:lng]) := group.groupdone(X_17:bat[:int]);                                           |
#     X_21:bat[:int] := algebra.projection(C_19:bat[:oid], X_17:bat[:int]);                                                          |
#     X_22:bat[:lng] := aggr.subcount(X_18:bat[:oid], X_18:bat[:oid], C_19:bat[:oid], false:bit);                                    |
#...
#     sql.resultSet(X_40:bat[:str], X_42:bat[:str], X_44:bat[:str], X_46:bat[:int], X_48:bat[:int], X_21:bat[:int], X_22:bat[:lng]); |
b:= bat.new(:int);
bat.append(b,0);
bat.append(b,1);
bat.append(b,2);
bat.append(b,3);
bat.append(b,4);
bat.append(b,5);
bat.append(b,6);
bat.append(b,7);
bat.append(b,8);
bat.append(b,9);
d:= bat.new(:int);
bat.append(d,0);
bat.append(d,1);
bat.append(d,1);
bat.append(d,2);
bat.append(d,2);
bat.append(d,3);
bat.append(d,3);
bat.append(d,4);
bat.append(d,4);
bat.append(d,5);
io.print(b,d);
(o,g,c):= group.groupdone(d);
io.print(o);
io.print(g,c);
j := algebra.projection(g,d);
io.print(j);
a:= aggr.subcount(o,o,g,false);
io.print(j,a);

(k1,k2,k3,k4):= partition.slice(d);
p1 := algebra.projection(k1,d);
p2 := algebra.projection(k2,d);
p3 := algebra.projection(k3,d);
p4 := algebra.projection(k4,d);
(o1,g1,c1):= group.groupdone(p1);
(o2,g2,c2):= group.groupdone(p2);
(o3,g3,c3):= group.groupdone(p3);
(o4,g4,c4):= group.groupdone(p4);
a1:= aggr.subcount(o1,o1,g1,false);
a2:= aggr.subcount(o2,o2,g2,false);
a3:= aggr.subcount(o3,o3,g3,false);
a4:= aggr.subcount(o4,o4,g4,false);
ma:= mat.pack(a1,a2,a3,a4);
j1 := algebra.projection(g1,k1);
j2 := algebra.projection(g2,k2);
j3 := algebra.projection(g3,k3);
j4 := algebra.projection(g4,k4);
l1 := algebra.projection(j1,d);
l2 := algebra.projection(j2,d);
l3 := algebra.projection(j3,d);
l4 := algebra.projection(j4,d);
ml := mat.pack(l1,l2,l3,l4);
m2 := mat.pack(j1,j2,j3,j4);
io.print(ml,ma);
