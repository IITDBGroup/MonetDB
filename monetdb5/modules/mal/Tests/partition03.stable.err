stderr of test 'partition03` in directory 'monetdb5/modules/mal` itself:


# 10:10:28 >  
# 10:10:28 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=39228" "--set" "mapi_usock=/var/tmp/mtest-23193/.s.monetdb.39228" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/home/mk/partition//Linux/var/MonetDB/mTests_monetdb5_modules_mal"
# 10:10:28 >  

# builtin opt 	gdk_dbpath = /export/scratch1/home/mk/partition//Linux/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 39228
# cmdline opt 	mapi_usock = /var/tmp/mtest-23193/.s.monetdb.39228
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /export/scratch1/home/mk/partition//Linux/var/MonetDB/mTests_monetdb5_modules_mal
# cmdline opt 	gdk_debug = 553648138

# 10:10:28 >  
# 10:10:28 >  "mclient" "-lmal" "-ftest" "-tnone" "-Eutf-8" "--host=/var/tmp/mtest-23193" "--port=39228"
# 10:10:28 >  


# 10:10:28 >  
# 10:10:28 >  "Done."
# 10:10:28 >  

