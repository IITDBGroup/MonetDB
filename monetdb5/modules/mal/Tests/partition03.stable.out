stdout of test 'partition03` in directory 'monetdb5/modules/mal` itself:


# 10:10:28 >  
# 10:10:28 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=39228" "--set" "mapi_usock=/var/tmp/mtest-23193/.s.monetdb.39228" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/home/mk/partition//Linux/var/MonetDB/mTests_monetdb5_modules_mal"
# 10:10:28 >  

# MonetDB 5 server v11.28.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 15.581 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2017 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://rocks073.scilens.private:39228/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-23193/.s.monetdb.39228
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.

# 10:10:28 >  
# 10:10:28 >  "mclient" "-lmal" "-ftest" "-tnone" "-Eutf-8" "--host=/var/tmp/mtest-23193" "--port=39228"
# 10:10:28 >  

#--------------------------#
# t	t	t  # name
# void	int	int  # type
#--------------------------#
[ 0@0,	0,	0	]
[ 1@0,	1,	1	]
[ 2@0,	2,	1	]
[ 3@0,	3,	2	]
[ 4@0,	4,	2	]
[ 5@0,	5,	3	]
[ 6@0,	6,	3	]
[ 7@0,	7,	4	]
[ 8@0,	8,	4	]
[ 9@0,	9,	5	]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0	]
[ 1@0,	1@0	]
[ 2@0,	1@0	]
[ 3@0,	2@0	]
[ 4@0,	2@0	]
[ 5@0,	3@0	]
[ 6@0,	3@0	]
[ 7@0,	4@0	]
[ 8@0,	4@0	]
[ 9@0,	5@0	]
#--------------------------#
# t	t	t  # name
# void	oid	lng  # type
#--------------------------#
[ 0@0,	0@0,	1	]
[ 1@0,	1@0,	2	]
[ 2@0,	3@0,	2	]
[ 3@0,	5@0,	2	]
[ 4@0,	7@0,	2	]
[ 5@0,	9@0,	1	]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	0	]
[ 1@0,	1	]
[ 2@0,	2	]
[ 3@0,	3	]
[ 4@0,	4	]
[ 5@0,	5	]
#--------------------------#
# t	t	t  # name
# void	int	lng  # type
#--------------------------#
[ 0@0,	0,	1	]
[ 1@0,	1,	2	]
[ 2@0,	2,	2	]
[ 3@0,	3,	2	]
[ 4@0,	4,	2	]
[ 5@0,	5,	1	]
#--------------------------#
# t	t	t  # name
# void	int	lng  # type
#--------------------------#
[ 0@0,	0,	1	]
[ 1@0,	4,	2	]
[ 2@0,	1,	2	]
[ 3@0,	5,	1	]
[ 4@0,	2,	2	]
[ 5@0,	3,	2	]

# 10:10:28 >  
# 10:10:28 >  "Done."
# 10:10:28 >  

